configure.ac

```diff
@@ -194,39 +192,45 @@ AS_CASE(["x$with_qtsingleapplication"],
 
 PKG_CHECK_MODULES(libtorrent,
                   [libtorrent-rasterbar >= 1.0.6],
-                  [CPPFLAGS="$libtorrent_CFLAGS $CPPFLAGS"
+                  [CXXFLAGS="$libtorrent_CFLAGS $CXXFLAGS"
                   LIBS="$libtorrent_LIBS $LIBS"])
 
 PKG_CHECK_MODULES(zlib,
                  [zlib >= 1.2.5.2],
-                 [CPPFLAGS="$zlib_CFLAGS $CPPFLAGS"
+                 [CXXFLAGS="$zlib_CFLAGS $CXXFLAGS"
                  LIBS="$zlib_LIBS $LIBS"])
 
-# Check compiler C++11 support
-AC_LANG_PUSH(C++)
-m4_define([DETECT_CPP11_PROGRAM],
-    [AC_LANG_PROGRAM([[
-#ifndef __cplusplus
-#error "This is not a C++ compiler"
-#elif __cplusplus < 201103L
-#error "This is not a C++11 compiler"
-#endif]],
-    [[]])])
-
-_tmp="$CXXFLAGS"
-CXXFLAGS="$CXXFLAGS -std=c++11"
+# Check if already in >= C++11 mode because of the flags returned by one of the above packages
+AC_MSG_CHECKING([if compiler is using C++11 or later mode])
 AC_COMPILE_IFELSE([DETECT_CPP11_PROGRAM()],
-    [],
-    [AC_MSG_ERROR([A compiler supporting C++11 is required.])])
-
-CXXFLAGS="$_tmp $libtorrent_CFLAGS"
-AC_COMPILE_IFELSE([DETECT_CPP11_PROGRAM()],
-    [],
-    [AC_MSG_ERROR([Compiler is not working in C++11 or later mode.
-    Make sure you use the same C++ mode for qBittorrent and its dependencies.
-    Example: `CXXFLAGS="\$CXXFLAGS -std=c++11" ./configure`])])
-CXXFLAGS="$_tmp"
-AC_LANG_POP([C++])
+                  [AC_MSG_RESULT([yes])
+                   QBT_CXX11_FOUND="yes"],
+                  [AC_MSG_RESULT([no])
+                   QBT_CXX11_FOUND="no"])
+                  
+# In case of no, check if the compiler can support at least C++11
+# and if yes, enable it leaving a warning to the user
+AS_IF([test "x$QBT_CXX11_FOUND" = "xno"],
+      [AC_MSG_CHECKING([if compiler supports C++11])
+       TMP_CXXFLAGS="$CXXFLAGS"
+       CXXFLAGS="$CXXFLAGS -std=c++11"
+       AC_COMPILE_IFELSE([DETECT_CPP11_PROGRAM()],
+                        [AC_MSG_RESULT([yes])
+                         AC_MSG_CHECKING([if C++11 is disabled by user defined compiler flags])
```

Comment:
Although you meant to check what users explicitly defined, it might also catch faults from dependency libraries right?
If so, then the description would be inaccurate.

Reply:
How about:
```
if C++11 is disabled by the set compiler flags
```
combined with this error message instead:

```
The compiler supports C++11 but the user or a dependency has explicitly enabled a lower mode.
```

?

Reply:
:+1: